nixos/modules/services/misc/packagekit.nix

```diff
@@ -33,26 +34,24 @@ in
           installing software. Software utilizing PackageKit can install
           software regardless of the package manager.
         '';
-    };
 
+      backend = mkOption {
+        type = types.enum [ "test_nop" ];
+        default = "test_nop"; # change to nix when https://github.com/NixOS/nix/issues/233 is done
+        description = ''
+          As the nix backend is incompatible with nix 2.0, the default is set to `test_nop`.
+        '';
+      };
+    };
   };
 
   config = mkIf cfg.enable {
 
-    services.dbus.packages = [ pkgs.packagekit ];
+    services.dbus.packages = with pkgs; [ packagekit ];
 
-    systemd.services.packagekit = {
-      description = "PackageKit Daemon";
-      wantedBy = [ "multi-user.target" ];
-      serviceConfig.ExecStart = "${pkgs.packagekit}/libexec/packagekitd";
-      serviceConfig.User = "root";
-      serviceConfig.BusName = "org.freedesktop.PackageKit";
-      serviceConfig.Type = "dbus";
-    };
+    systemd.packages = with pkgs; [ packagekit ];
```

Comment:
We still need to override the `ExecStart` of said unit with
```
systemd.services.packagekit.serviceConfig.ExecStart = "${pkgs.packagekit}/libexec/packagekitd --backend ${cfg.backend}";
```

Reply:
No, because now we write the backend to the config file instead.

Reply:
Ooh didn't catch that :+1: